tutorials/xr/xr_action_map.rst

```diff
@@ -0,0 +1,246 @@
+ * grab, which detects the user wants to hold something.
+ * shoot, which detects the user wants to fire the weapon they are holding.
+ * haptic, which allows us to output haptic feedback.
+
+Now note that we don't make a difference between the left and right hand. This is something that is determined at the next stage. We've implemented the action system in such a way that you can bind the same action to both hands, and the correct :ref:`XRController3D <class_xrcontroller3d>` node will emit the signal.
+
+.. warning::
+  Note that for both grab and shoot we've used the ``Bool`` type. As mentioned before, OpenXR does automatic conversions from an analogue controls however not all XR Runtimes currently apply sensible thresholds.
+
+  We recommend as a workaround to use the ``Float`` type when interacting with triggers and grip buttons and apply your own threshold.
+
+  For A/B/X/Y buttons the ``Bool`` type works fine.
+
+Now that we have our basic actions defined, it's time to hook them up.
+
+Profiles
+========
+
+In OpenXR controller bindings are captured in what are called "Interaction Profiles", we've simply shortened it to "Profiles" because it takes up less space.
+
+This generic name is chosen because controllers don't cover the entire system. Currently there are also profiles for trackers, remotes and tracked pens. There are also provisions for devices such as treadmills, haptic vests and such even though those are not part of the specification yet.
+
+.. warning::
+  It is important to know that OpenXR has strict checking on supported devices. The core specification identifies a number of controllers and similar devices with their supported inputs and outputs, every XR runtime must accept these interaction profiles even if they aren't applicable. New devices are added through extensions and XR runtimes must specify which they support. XR runtimes that do not support a device added through extensions will not accept these profiles. XR runtimes that do not support added input or output types will often crash if supplied.
+
+  As such Godot keeps meta data of all available devices, their inputs and outputs and which extension adds support for them. You can create interaction profiles for all devices you wish to support, Godot will filter out those not supported by the XR runtime the user is using.
+
+  This does mean that in order to support new devices, software changes are needed.
+
+It is however also important to note that the action map has been designed with this in mind. When new devices enter the market, or when your users use devices that you do not have access to, the action map system puts the onus with the XR runtime. It is the XR runtimes job to choose the best fitting interaction profile that has been specified and adapt it for the controller the user is using.
+
+How the XR runtime does this is left to the implementation of the runtime and there are thus vast differences between the runtimes.
```

Comment:
```suggestion
How the XR runtime does this is left to the implementation of the runtime and there are thus vast differences between the runtimes. Some runtimes might even permit users to edit the bindings themselves.
```

(You included this in a warning later on, but I feel this makes sense to mention here.)